realm/rfc/0001-backlink-query-syntax.md

```diff
@@ -0,0 +1,232 @@
+- Feature Name: Inverse relationship syntax
+- Start Date: 2016-05-30
+- RFC PR: 
+- Version: 3
+
+# Summary
+
+It should be possible to define and query backlinks in both `Realm` and 
+`DynamicRealm`. As a `DynamicRealm` can work without _any_ Realm model 
+classes, it should be possible to query backlinks without them being defined there.
+
+# Motivation
+
+Backlinks are special in Realm. They are not defined explicitly in the schema,
+but are automatically created when defining Links and LinkLists (forward links).
+
+This creates some conflicts with the current query system as it rely on all 
+fields being defined by the schema. This is called *named backlinks*.
+
+This proposal introduces a new syntax for queries so it is possible to query
+across a backlink even though it hasn't been defined by a model class. This is
+called *unnamed backlinks*.
```

Comment:
Sure. Primary reason is Migrations where you don't have Realm model classes at all and in Java you can also access a Realm in "full dynamic mode", where no Realm classes needs to be defined either. This is e.g. used by browsers and other libraries that work with data in a largely dynamic fashion (CSV import ).